lib/ConvoyLeader/src/App.cpp

```diff
@@ -60,10 +61,19 @@
 
 void App::setup()
 {
+    if (false == Board::getInstance().init())

```

Comment:
It would be better if Initialization of the HAL doesn't fail. I see that its because the SocketClient init() will do the connection establishment. But wouldn't it be better to have the connection establishment with possible reconnect behaviour during processing?

Reply:
The connection of the SocketClient has been moved to the process rutine. However part of the initialization can still fail, for example, the context creation. Other components can fail, for example, the battery. For this reason, the HAL should fail in the setup.

This makes also sense in the Target environment, where the USB Host Chip can fail to initialize, or WiFi can fail to connect.